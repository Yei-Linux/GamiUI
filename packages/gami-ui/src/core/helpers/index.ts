export * from './date-formatter'
export * from './validations.helper'
