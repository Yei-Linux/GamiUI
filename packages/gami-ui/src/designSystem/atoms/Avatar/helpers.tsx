import { TextModeType } from 'core/domain/types'
import { avatarTextModes } from './constants'
import { REGEX_RULES } from 'core/utils/constants'

export const validateMaxLetters = (
  firstLettersGot: string,
  maxSizeLetters: number
) => {
  const size = firstLettersGot.length

  if (size > maxSizeLetters) {
    return firstLettersGot.substring(0, maxSizeLetters)
  }

  return firstLettersGot
}

export const showOnlyFirstLettersOnText = (
  text: string,
  type: TextModeType,
  maxSizeLetters: number
): string => {
  const regexType = avatarTextModes[type]
  const firstLettersCaptured: string[] | null = text.match(
    REGEX_RULES[regexType]
  )

  const joinFirstLetters = firstLettersCaptured?.join('') || ''

  return validateMaxLetters(joinFirstLetters, maxSizeLetters)
}
