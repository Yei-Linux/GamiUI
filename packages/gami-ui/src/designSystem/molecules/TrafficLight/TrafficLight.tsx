import 'regenerator-runtime/runtime'

import React, { useEffect, useState } from 'react'
import * as S from './TrafficLight.styles'

import { IGeneralProps } from 'core/domain/interfaces/IGeneralProps'
import { timeout } from 'core/helpers/utilities.helper'
import { TrafficLightColors } from './constants'

export type TrafficLightProps = IGeneralProps

const TrafficLight = ({ ...args }: TrafficLightProps) => {
  const [counter, setCounter] = useState(10)
  const [isBrightRed, setIsBrightRed] = useState(true)
  const [isBrightOrange, setIsBrightOrange] = useState(false)
  const [isBrightGreen, setIsBrightGreen] = useState(false)

  const startingCountdownLight = async (): Promise<void> => {
    await timeout(3000)
    setIsBrightRed(false)
    setIsBrightOrange(true)
    await timeout(3000)
    setIsBrightOrange(false)
    setIsBrightGreen(true)
    await timeout(4000)
    setIsBrightGreen(false)
  }

  const startingCountdownTimer = async (): Promise<void> => {
    for (const item of [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]) {
      await timeout(1000)
      setCounter(item)
    }
  }

  useEffect(() => {
    startingCountdownLight()
    startingCountdownTimer()
  }, [])

  return (
    <S.TrafficLight {...args}>
      <S.TrafficLightContainer>
        <S.Balls
          $color={TrafficLightColors.red[isBrightRed ? 'brightColor' : 'color']}
          $bright={isBrightRed ? TrafficLightColors.red['shadow'] : 'none'}
        />
        <S.Balls
          $color={
            TrafficLightColors.orange[isBrightOrange ? 'brightColor' : 'color']
          }
          $bright={
            isBrightOrange ? TrafficLightColors.orange['shadow'] : 'none'
          }
        />
        <S.Balls
          $color={
            TrafficLightColors.green[isBrightGreen ? 'brightColor' : 'color']
          }
          $bright={isBrightGreen ? TrafficLightColors.green['shadow'] : 'none'}
        />
      </S.TrafficLightContainer>

      <S.TextCounter>
        <p>{counter}</p>
      </S.TextCounter>
    </S.TrafficLight>
  )
}

export default TrafficLight
