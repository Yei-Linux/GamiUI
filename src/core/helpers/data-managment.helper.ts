export const mapEnum = (enumerable: any): any[] => {
  const enumMembers: any[] = Object.keys(enumerable).map(
    (key) => enumerable[key]
  )
  return enumMembers
}
