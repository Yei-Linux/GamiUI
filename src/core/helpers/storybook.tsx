import React from "react";
import TableStories from "../../styles/utilities/TableStories";
import { options as optionsStyle } from "../utils/constants";

export const getTemplate = (Component: React.ComponentType<any>): any => (
  args: any
) => {
  return <Component {...args} />;
};

export const getListTemplate = (Component: React.ComponentType<any>): any => ({
  items,
  field,
  ...args
}: {
  items: any[];
  field: any;
}) =>
  items.map((item, index) => (
    <TableStories key={index} item={item} field={field}>
      <Component {...args} {...item} />
    </TableStories>
  ));

export const stylesControl = {
  style: { control: "object" },
  textAlign: { control: { type: "select", options: optionsStyle.textAlign } },
  fontWeight: {
    control: { type: "select", options: optionsStyle.fontWeight },
  },
  width: { control: { type: "select", options: optionsStyle.width } },
  shadow: { control: { type: "select", options: optionsStyle.shadow } },
  border: { control: { type: "select", options: optionsStyle.border } },
  heigth: { control: { type: "select", options: optionsStyle.height } },
};