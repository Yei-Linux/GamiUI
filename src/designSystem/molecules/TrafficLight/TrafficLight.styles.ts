import styled from "@emotion/styled";
import tw from "twin.macro";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const TrafficLightWrapper = styled.div<{
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
}>`
  ${tw`flex flex-col justify-center items-center w-fit rounded-md`}
  padding: 10px 0px;
  background-color: #3a3a3a;

  ${(props: any) => twinStyles(props)};
`;

export const TrafficLightContainer = tw.div`flex flex-col justify-center items-center w-full h-full`;

export const Balls = styled.div<{ color: string; bright: string }>`
  border-radius: 4em;
  height: 40px;
  width: 40px;
  margin: 5px 30px;

  box-shadow: ${(props) => props.bright};
  background-color: ${(props) => props.color};
`;

export const TextCounter = styled.div`
  ${tw`flex justify-center items-center w-full rounded-md font-bold`}
  min-height: 50px;
  margin: 15px 0px;
  color: #bababa;
  font-size: 3em;
  transition: color 0.3s;
`;
