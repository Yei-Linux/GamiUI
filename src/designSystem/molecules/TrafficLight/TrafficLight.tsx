import "regenerator-runtime/runtime";
import React, { useEffect, useState } from "react";
import {
  Balls,
  TextCounter,
  TrafficLightContainer,
  TrafficLightWrapper,
} from "./TrafficLight.styles";
import { TrafficLightColors } from "../../../core/utils/trafficlight.util";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";

export interface TrafficLightProps extends IGeneralProps {}

const TrafficLight = ({ ...args }: TrafficLightProps) => {
  const [counter, setCounter] = useState(10);
  const [isBrightRed, setIsBrightRed] = useState(true);
  const [isBrightOrange, setIsBrightOrange] = useState(false);
  const [isBrightGreen, setIsBrightGreen] = useState(false);

  const timeout = (t: number): Promise<any> => {
    return new Promise((resolve, _reject) => {
      setTimeout(resolve, t);
    });
  };

  const startingCountdownLight = async (): Promise<void> => {
    await timeout(3000);
    setIsBrightRed(false);
    setIsBrightOrange(true);
    await timeout(3000);
    setIsBrightOrange(false);
    setIsBrightGreen(true);
    await timeout(4000);
    setIsBrightGreen(false);
  };

  const startingCountdownTimer = async (): Promise<void> => {
    for (let item of [9, 8, 7, 6, 5, 4, 3, 2, 1, 0]) {
      await timeout(1000);
      setCounter(item);
    }
  };

  useEffect(() => {
    startingCountdownLight();
    startingCountdownTimer();
  }, []);

  return (
    <TrafficLightWrapper {...args}>
      <TrafficLightContainer>
        <Balls
          color={TrafficLightColors.red[isBrightRed ? "brightColor" : "color"]}
          bright={isBrightRed ? TrafficLightColors.red["shadow"] : "none"}
        />
        <Balls
          color={
            TrafficLightColors.orange[isBrightOrange ? "brightColor" : "color"]
          }
          bright={isBrightOrange ? TrafficLightColors.orange["shadow"] : "none"}
        />
        <Balls
          color={
            TrafficLightColors.green[isBrightGreen ? "brightColor" : "color"]
          }
          bright={isBrightGreen ? TrafficLightColors.green["shadow"] : "none"}
        />
      </TrafficLightContainer>

      <TextCounter>
        <p>{counter}</p>
      </TextCounter>
    </TrafficLightWrapper>
  );
};

TrafficLight.defaultProps = {
  shadow: "MEDIUM",
};

export default TrafficLight;
