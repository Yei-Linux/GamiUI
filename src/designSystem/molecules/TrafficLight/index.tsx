export { default } from "./TrafficLight";
export { options } from "./constants";
