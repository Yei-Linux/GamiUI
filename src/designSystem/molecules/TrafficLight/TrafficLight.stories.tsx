import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import TrafficLight from ".";

const Template = getTemplate(TrafficLight);

export default {
  title: "Molecules/TrafficLight",
  component: TrafficLight,
  args: { ...TrafficLight.defaultProps },
  argTypes: {
    ...stylesControl,
  },
} as Meta;

export const Basic = Template.bind({});
Basic.args = {};
