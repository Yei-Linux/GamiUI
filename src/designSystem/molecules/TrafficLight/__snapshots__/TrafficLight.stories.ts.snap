// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/TrafficLight Basic 1`] = `
<div
  className="css-1fm3r8h-TrafficLightWrapper e4ce4rr0"
>
  <div
    className="css-7ya29i-TrafficLightContainer e4ce4rr1"
  >
    <div
      className="css-15iids2-Balls e4ce4rr2"
      color="#c33"
    />
    <div
      className="css-pha6og-Balls e4ce4rr2"
      color="#9e6d1d"
    />
    <div
      className="css-1vo0czq-Balls e4ce4rr2"
      color="#4c751e"
    />
  </div>
  <div
    className="css-9sid66-TextCounter e4ce4rr3"
  >
    <p>
      10
    </p>
  </div>
</div>
`;
